src/odemis/driver/test/pwrmccdaq_test.py

```diff
@@ -0,0 +1,175 @@
+        self.assertTrue(isinstance(self.mcc_device.device, pwrmccdaq.MCCDeviceSimulator))
+        # check if the status trigger of the VA is False
+        self.assertFalse(self.mcc_device.interlockTriggered.value)
+
+        old_bit_status = self.mcc_device.device.DIn(port)
+        # be sure all the bit values of the DI port are 1 at the start
+        self.assertTrue(old_bit_status == 0xff)
+
+        # trigger the VA by setting the TTL signal on the right port bit to 0 -> hex 251
+        self.mcc_device.device.DOut(port, 0xfb)
+        time.sleep(pwrmccdaq.INTERLOCK_POLL_INTERVAL)  # wait just long enough until the next thread's interval
+        self.assertTrue(self.mcc_device.interlockTriggered.value)
+        # check if the bit status is now the new bit status
+        new_bit_status = self.mcc_device.device.DIn(port)
+        self.assertNotEqual(old_bit_status, new_bit_status)
+
+    def test_incorrect_di_list(self):
+        """
+        Test for the instantiation of a basic MCC device with incorrect DI list values or structure.
+        Also does a small check for the poll thread to not have started when there are no status
+        DI's to check.
+        """
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {2: ["interlockTriggered", False], 99: ["testPropery", True]}
+        with self.assertRaises(ValueError):
+            pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {11: [True, "testPropery"]}
+        with self.assertRaises(TypeError):
+            pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # check list with no channel values given
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = None
+        self.mcc_device = pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # check if there are no extra Boolean VA's
+        for va in model.getVAs(self.mcc_device):
+            self.assertFalse(isinstance(va, model.BooleanVA))
+
+        # be sure that there is no poll tread started as there are no Boolean VA's to check
+        self.assertFalse(self.mcc_device._status_thread)
+
+    def test_thread_termination(self):
+        """
+        Check if the polling thread for the status changes of the DI's is working properly.
+        """
+        self.mcc_device = pwrmccdaq.MCCDevice(**CONFIG_MCC)
+
+        # check if polling of the DI port bits has started
+        self.assertTrue(self.mcc_device._status_thread.is_alive())
+        self.assertFalse(self.mcc_device._status_thread.terminated)
+
+        self.mcc_device.terminate()
+        time.sleep(0.5)  # give the thread some time to suspend
+
+        # check if polling of the DI port bits has suspended
+        self.assertTrue(self.mcc_device._status_thread.terminated)
+        self.assertFalse(self.mcc_device._status_thread.is_alive())
+
+    def tearDown(self):
+        if self.mcc_device._status_thread:
```

Comment:
Drop that `if`. It should always be safe to call `.terminate()`. If it's not, the MCCDevice component should be fixed.